libcaf_core/caf/flow/observable.test.cpp

```diff
@@ -249,6 +249,83 @@ TEST("on_error_complete() suppresses errors") {
+    }
+    SECTION("observable") {
+      check_eq(collect(mat(range(1, 1)).on_error_return_item(42)), vector{1});
+      check_eq(collect(mat(range(1, 2)).on_error_return_item(42)),
+               vector{1, 2});
+      check_eq(collect(mat(range(1, 3)).on_error_return_item(42)),
+               vector{1, 2, 3});
+      check_eq(collect(mat(range(1, 0)).on_error_return_item(42)), nil);
+    }
+  }
+  SECTION("on_error_return_item() returns item when an error occurs") {
+    SECTION("blueprint") {
+      check_eq(collect(range(1, 0)
+                         .concat(obs_error(sec::runtime_error))
+                         .on_error_return_item(42)),
+               vector{42});
+      check_eq(collect(range(1, 2)
+                         .concat(obs_error(sec::runtime_error))
+                         .concat(range(1, 2))
+                         .on_error_return_item(42)),
+               vector{1, 2, 42});
+      check_eq(collect(range(1, 3)
+                         .concat(obs_error(sec::runtime_error))
+                         .concat(range(1, 3))
+                         .on_error_return_item(42)),
+               vector{1, 2, 3, 42});
+    }
+    SECTION("observable") {
+      check_eq(collect(mat(range(1, 0).concat(obs_error(sec::runtime_error)))
+                         .on_error_return_item(42)),
+               vector{42});
+      check_eq(collect(mat(range(1, 2)
+                             .concat(obs_error(sec::runtime_error))
+                             .concat(range(1, 2)))
+                         .on_error_return_item(42)),
+               vector{1, 2, 42});
+      check_eq(collect(mat(range(1, 3)
+                             .concat(obs_error(sec::runtime_error))
+                             .concat(range(1, 3)))
+                         .on_error_return_item(42)),
+               vector{1, 2, 3, 42});
+    }
+  }
+  SECTION(
+    "on_error_return_item() forwards error when an item can not be returned") {
+    SECTION("blueprint") {
+      check_eq(collect(range(1, 2)
+                         .concat(obs_error(sec::runtime_error))
+                         .on_error_return_item(42)
+                         .take(2)),
+               make_error(sec::runtime_error));
+      check_eq(collect(range(1, 3)
+                         .concat(obs_error(sec::runtime_error))
+                         .on_error_return_item(42)
+                         .take(3)),
+               make_error(sec::runtime_error));
+    }
+    SECTION("observable") {
+      check_eq(collect(mat(range(1, 2).concat(obs_error(sec::runtime_error)))
+                         .on_error_return_item(42)
+                         .take(2)),
+               make_error(sec::runtime_error));
+      check_eq(collect(mat(range(1, 3).concat(obs_error(sec::runtime_error)))
+                         .on_error_return_item(42)
+                         .take(3)),
+               make_error(sec::runtime_error));
+    }
+  }
```

Comment:
The sequence must always behave the same, whether or not we cut it short. Merge the tests into the sections above (and they shouldn't result in an error).